pkg/cloud/services/networking/securitygroups.go

```diff
@@ -56,14 +56,27 @@ func (s *Service) ReconcileSecurityGroups(clusterName string, openStackCluster *
 		s.logger.V(4).Info("No need to reconcile security groups", "cluster", clusterName)
 		return nil
 	}
+
+	secControlPlaneGroupName := fmt.Sprintf("%s-cluster-%s-secgroup-%s", secGroupPrefix, clusterName, controlPlaneSuffix)
```

Comment:
this name is calculated here and in generateControlPlaneGroup and in generateWorkerGroup similar holds for the worker group name. Can we make sure we have only one place where we generate the sec group name and then hand it over?

Reply:
Generating the sec group names is now only once at the beginning of ReconcileSecurityGroups.